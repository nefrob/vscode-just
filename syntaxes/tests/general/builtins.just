# Reserved keywords

alias
set
export
import
mod

# Control keywords

if
else

# Builtin functions

# System info

arch()
num_cpus()
os()
os_family()

# External commands

shell(command, arg1)

# Environment variables

env_var(key)
env_var_or_default(key, default)
env(key)
env(key, default)

# Invocation information

is_dependency()

# Invocation directory

invocation_directory()
invocation_dir()
invocation_directory_native()
invocation_dir_native()

# Just

justfile()
justfile_directory()
justfile_dir()
just_executable()
just_pid()

# Source

source_file()
source_directory()
source_dir()
module_file()
module_directory()
module_dir()


# Strings

append(suffix, s)
prepend(prefix, s)
encode_uri_component(s)
quote(s)
replace(s, from, to) 
replace_regex(s, regex, replacement)
trim(s)
trim_end(s)
trim_end_match(s, pat)
trim_end_matches(s, pat) 
trim_start(s) 
trim_start_match(s, pat)
trim_start_matches(s, pat)

capitalize(s)
kebabcase(s)
lowercamelcase(s)
lowercase(s)
shoutykebabcase(s)
shoutysnakecase(s)
snakecase(s)
titlecase(s)
uppercamelcase(s)
uppercase(s)

# Paths

absolute_path(path)
canonicalize(path)
extension(path)
file_name(path)
file_stem(path)
parent_directory(path)
parent_dir()
without_extension(path)

clean(path)
join(a, b)

path_exists(path)

# Errors

error(message)
assert(condition, expressions)

# Utils

blake3(string)
blake3_file(path)
sha256(string)
sha256_file(path)
uuid()

choose(n, alphabet)

datetime(format)
datetime_utc(format)

semver_matches(version, requirement)

cache_directory()
cache_dir()
config_directory()
config_dir()
config_local_directory()
config_local_dir()
data_directory()
data_dir()
data_local_directory()
data_local_dir()
executable_directory()
executable_dir()
home_directory()
home_dir()
