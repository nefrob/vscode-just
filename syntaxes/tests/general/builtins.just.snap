># Reserved keywords
#^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>alias
#^^^^^ source.just keyword.other.reserved.just
>set
#^^^ source.just keyword.other.reserved.just
>export
#^^^^^^ source.just keyword.other.reserved.just
>import
#^^^^^^ source.just keyword.other.reserved.just
>mod
#^^^ source.just keyword.other.reserved.just
>
># Control keywords
#^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>if
#^^ source.just keyword.control.conditional.just
>else
#^^^^ source.just keyword.control.conditional.just
>
># Builtin functions
#^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
># System info
#^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>arch()
#^^^^ source.just support.function.builtin.just
#    ^ source.just
#     ^ source.just
>num_cpus()
#^^^^^^^^ source.just support.function.builtin.just
#        ^ source.just
#         ^ source.just
>os()
#^^ source.just support.function.builtin.just
#  ^ source.just
#   ^ source.just
>os_family()
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^ source.just
>
># External commands
#^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>shell(command, arg1)
#^^^^^ source.just support.function.builtin.just
#     ^ source.just
#      ^^^^^^^^^^^^^ source.just
#                   ^ source.just
>
># Environment variables
#^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>env_var(key)
#^^^^^^^ source.just support.function.builtin.just
#       ^ source.just
#        ^^^ source.just
#           ^ source.just
>env_var_or_default(key, default)
#^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                  ^ source.just
#                   ^^^^^^^^^^^^ source.just
#                               ^ source.just
>env(key)
#^^^ source.just support.function.builtin.just
#   ^ source.just
#    ^^^ source.just
#       ^ source.just
>env(key, default)
#^^^ source.just support.function.builtin.just
#   ^ source.just
#    ^^^^^^^^^^^^ source.just
#                ^ source.just
>
># Invocation information
#^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>is_dependency()
#^^^^^^^^^^^^^ source.just support.function.builtin.just
#             ^ source.just
#              ^ source.just
>
># Invocation directory
#^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>invocation_directory()
#^^^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                    ^ source.just
#                     ^ source.just
>invocation_dir()
#^^^^^^^^^^^^^^ source.just support.function.builtin.just
#              ^ source.just
#               ^ source.just
>invocation_directory_native()
#^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                           ^ source.just
#                            ^ source.just
>invocation_dir_native()
#^^^^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                     ^ source.just
#                      ^ source.just
>
># Just
#^^^^^^ source.just comment.line.number-sign.just
>
>justfile()
#^^^^^^^^ source.just support.function.builtin.just
#        ^ source.just
#         ^ source.just
>justfile_directory()
#^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                  ^ source.just
#                   ^ source.just
>justfile_dir()
#^^^^^^^^^^^^ source.just support.function.builtin.just
#            ^ source.just
#             ^ source.just
>just_executable()
#^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#               ^ source.just
#                ^ source.just
>just_pid()
#^^^^^^^^ source.just support.function.builtin.just
#        ^ source.just
#         ^ source.just
>
># Source
#^^^^^^^^ source.just comment.line.number-sign.just
>
>source_file()
#^^^^^^^^^^^ source.just support.function.builtin.just
#           ^ source.just
#            ^ source.just
>source_directory()
#^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                ^ source.just
#                 ^ source.just
>source_dir()
#^^^^^^^^^^ source.just support.function.builtin.just
#          ^ source.just
#           ^ source.just
>module_file()
#^^^^^^^^^^^ source.just support.function.builtin.just
#           ^ source.just
#            ^ source.just
>module_directory()
#^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                ^ source.just
#                 ^ source.just
>module_dir()
#^^^^^^^^^^ source.just support.function.builtin.just
#          ^ source.just
#           ^ source.just
>
>
># Strings
#^^^^^^^^^ source.just comment.line.number-sign.just
>
>append(suffix, s)
#^^^^^^ source.just support.function.builtin.just
#      ^ source.just
#       ^^^^^^^^^ source.just
#                ^ source.just
>prepend(prefix, s)
#^^^^^^^ source.just support.function.builtin.just
#       ^ source.just
#        ^^^^^^^^^ source.just
#                 ^ source.just
>encode_uri_component(s)
#^^^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                    ^ source.just
#                     ^ source.just
#                      ^ source.just
>quote(s)
#^^^^^ source.just support.function.builtin.just
#     ^ source.just
#      ^ source.just
#       ^ source.just
>replace(s, from, to) 
#^^^^^^^ source.just support.function.builtin.just
#       ^ source.just
#        ^^^^^^^^^^^ source.just
#                   ^ source.just
#                    ^^ source.just
>replace_regex(s, regex, replacement)
#^^^^^^^^^^^^^ source.just support.function.builtin.just
#             ^ source.just
#              ^^^^^^^^^^^^^^^^^^^^^ source.just
#                                   ^ source.just
>trim(s)
#^^^^ source.just support.function.builtin.just
#    ^ source.just
#     ^ source.just
#      ^ source.just
>trim_end(s)
#^^^^^^^^ source.just support.function.builtin.just
#        ^ source.just
#         ^ source.just
#          ^ source.just
>trim_end_match(s, pat)
#^^^^^^^^^^^^^^ source.just support.function.builtin.just
#              ^ source.just
#               ^^^^^^ source.just
#                     ^ source.just
>trim_end_matches(s, pat) 
#^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                ^ source.just
#                 ^^^^^^ source.just
#                       ^ source.just
#                        ^^ source.just
>trim_start(s) 
#^^^^^^^^^^ source.just support.function.builtin.just
#          ^ source.just
#           ^ source.just
#            ^ source.just
#             ^^ source.just
>trim_start_match(s, pat)
#^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                ^ source.just
#                 ^^^^^^ source.just
#                       ^ source.just
>trim_start_matches(s, pat)
#^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                  ^ source.just
#                   ^^^^^^ source.just
#                         ^ source.just
>
>capitalize(s)
#^^^^^^^^^^ source.just support.function.builtin.just
#          ^ source.just
#           ^ source.just
#            ^ source.just
>kebabcase(s)
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^ source.just
#           ^ source.just
>lowercamelcase(s)
#^^^^^^^^^^^^^^ source.just support.function.builtin.just
#              ^ source.just
#               ^ source.just
#                ^ source.just
>lowercase(s)
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^ source.just
#           ^ source.just
>shoutykebabcase(s)
#^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#               ^ source.just
#                ^ source.just
#                 ^ source.just
>shoutysnakecase(s)
#^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#               ^ source.just
#                ^ source.just
#                 ^ source.just
>snakecase(s)
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^ source.just
#           ^ source.just
>titlecase(s)
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^ source.just
#           ^ source.just
>uppercamelcase(s)
#^^^^^^^^^^^^^^ source.just support.function.builtin.just
#              ^ source.just
#               ^ source.just
#                ^ source.just
>uppercase(s)
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^ source.just
#           ^ source.just
>
># Paths
#^^^^^^^ source.just comment.line.number-sign.just
>
>absolute_path(path)
#^^^^^^^^^^^^^ source.just support.function.builtin.just
#             ^ source.just
#              ^^^^ source.just
#                  ^ source.just
>canonicalize(path)
#^^^^^^^^^^^^ source.just support.function.builtin.just
#            ^ source.just
#             ^^^^ source.just
#                 ^ source.just
>extension(path)
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^^^^ source.just
#              ^ source.just
>file_name(path)
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^^^^ source.just
#              ^ source.just
>file_stem(path)
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^^^^ source.just
#              ^ source.just
>parent_directory(path)
#^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                ^ source.just
#                 ^^^^ source.just
#                     ^ source.just
>parent_dir()
#^^^^^^^^^^ source.just support.function.builtin.just
#          ^ source.just
#           ^ source.just
>without_extension(path)
#^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                 ^ source.just
#                  ^^^^ source.just
#                      ^ source.just
>
>clean(path)
#^^^^^ source.just support.function.builtin.just
#     ^ source.just
#      ^^^^ source.just
#          ^ source.just
>join(a, b)
#^^^^ source.just support.function.builtin.just
#    ^ source.just
#     ^^^^ source.just
#         ^ source.just
>
>path_exists(path)
#^^^^^^^^^^^ source.just support.function.builtin.just
#           ^ source.just
#            ^^^^ source.just
#                ^ source.just
>
># Errors
#^^^^^^^^ source.just comment.line.number-sign.just
>
>error(message)
#^^^^^ source.just support.function.builtin.just
#     ^ source.just
#      ^^^^^^^ source.just
#             ^ source.just
>assert(condition, expressions)
#^^^^^^ source.just support.function.builtin.just
#      ^ source.just
#       ^^^^^^^^^^^^^^^^^^^^^^ source.just
#                             ^ source.just
>
># Utils
#^^^^^^^ source.just comment.line.number-sign.just
>
>blake3(string)
#^^^^^^ source.just support.function.builtin.just
#      ^ source.just
#       ^^^^^^ source.just
#             ^ source.just
>blake3_file(path)
#^^^^^^^^^^^ source.just support.function.builtin.just
#           ^ source.just
#            ^^^^ source.just
#                ^ source.just
>sha256(string)
#^^^^^^ source.just support.function.builtin.just
#      ^ source.just
#       ^^^^^^ source.just
#             ^ source.just
>sha256_file(path)
#^^^^^^^^^^^ source.just support.function.builtin.just
#           ^ source.just
#            ^^^^ source.just
#                ^ source.just
>uuid()
#^^^^ source.just support.function.builtin.just
#    ^ source.just
#     ^ source.just
>
>choose(n, alphabet)
#^^^^^^ source.just support.function.builtin.just
#      ^ source.just
#       ^^^^^^^^^^^ source.just
#                  ^ source.just
>
>datetime(format)
#^^^^^^^^ source.just support.function.builtin.just
#        ^ source.just
#         ^^^^^^ source.just
#               ^ source.just
>datetime_utc(format)
#^^^^^^^^^^^^ source.just support.function.builtin.just
#            ^ source.just
#             ^^^^^^ source.just
#                   ^ source.just
>
>semver_matches(version, requirement)
#^^^^^^^^^^^^^^ source.just support.function.builtin.just
#              ^ source.just
#               ^^^^^^^^^^^^^^^^^^^^ source.just
#                                   ^ source.just
>
>style(name)
#^^^^^ source.just support.function.builtin.just
#     ^ source.just
#      ^^^^ source.just
#          ^ source.just
>
>cache_directory()
#^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#               ^ source.just
#                ^ source.just
>cache_dir()
#^^^^^^^^^ source.just support.function.builtin.just
#         ^ source.just
#          ^ source.just
>config_directory()
#^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                ^ source.just
#                 ^ source.just
>config_dir()
#^^^^^^^^^^ source.just support.function.builtin.just
#          ^ source.just
#           ^ source.just
>config_local_directory()
#^^^^^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                      ^ source.just
#                       ^ source.just
>config_local_dir()
#^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                ^ source.just
#                 ^ source.just
>data_directory()
#^^^^^^^^^^^^^^ source.just support.function.builtin.just
#              ^ source.just
#               ^ source.just
>data_dir()
#^^^^^^^^ source.just support.function.builtin.just
#        ^ source.just
#         ^ source.just
>data_local_directory()
#^^^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                    ^ source.just
#                     ^ source.just
>data_local_dir()
#^^^^^^^^^^^^^^ source.just support.function.builtin.just
#              ^ source.just
#               ^ source.just
>executable_directory()
#^^^^^^^^^^^^^^^^^^^^ source.just support.function.builtin.just
#                    ^ source.just
#                     ^ source.just
>executable_dir()
#^^^^^^^^^^^^^^ source.just support.function.builtin.just
#              ^ source.just
#               ^ source.just
>home_directory()
#^^^^^^^^^^^^^^ source.just support.function.builtin.just
#              ^ source.just
#               ^ source.just
>home_dir()
#^^^^^^^^ source.just support.function.builtin.just
#        ^ source.just
#         ^ source.just
>
>which()
#^^^^^ source.just support.function.builtin.just
#     ^ source.just
#      ^ source.just
>require()
#^^^^^^^ source.just support.function.builtin.just
#       ^ source.just
#        ^ source.just
>read(path)
#^^^^ source.just support.function.builtin.just
#    ^ source.just
#     ^^^^ source.just
#         ^ source.just
>